### DISCUSSION TITLE
network/vxlan

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#225] [General]
# network/vxlan

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/network/vxlan.html

**Comment by Do1122** [comment]
因此以 VXLAN 为代表的 Overlay 网络作为一种不调底层网络实现的容器组网技术，快速的在容器领域铺开了。
这里”不调底层“是指？

**Reply by isno** [reply]
无需调整底层网络（也就是物理网络）。

我这边再改一改

**Comment by partnerSun** [comment]
vxlan是一种overlay技术，vxlan是在underlay的基础之建立上的，建立vxlan 的前提是底层网络可以通信，而vxlan的作用就是把底层的网络封装成一个大二层，这么理解对么

**Reply by isno** [reply]
没错，就是这么回事。

**Comment by madeinchinalmc** [comment]
老师你好，VLAN的单臂路由模式工作原理没太看清楚，我读完后按照我的理解归纳了下工作原理。单臂路由工作流程为：
VLAN A 主机A1发出数据包（目的ip是VLAN B的主机B1  172.16.2.2/24），因为A和B不在一个子网，所以数据经由交换机后，没办法直接广播至B子网。
引用单臂路由模式后，此时A1的数据发出后将流向路由器，路由器提前配置好子接口，将A1的数据包转向B1所在的子网B。
这个过程中，VLAN A 和 VLAN B 的子接口应该有一个映射关系，这个映射关系是存在路由器的路由表吗？ A1的数据包流向路由器后，会修改原数据包的一些信息吗（例如以太帧报文头中的VLAN Tag的VLAN ID改为B子网的VLAN ID）？

**Reply by isno** [reply]
你说得没错。
你自己在网上搜一下 单臂路由配置，逻辑接口里面有 vlan id 还有 子网设置之类。

**Comment by bingofather** [comment]
你好博主，这些图你是用什么软件画的

**Reply by isno** [reply]
google drawing

**Comment by lvdeshuii** [comment]
二层以太帧是 -> 二层以太帧时

**Comment by zjyyzhengli** [comment]
文中关于24 bytes VNI字段的描述，应该是24 **bits**。
还有图 3-17 中“路由器“侧的“子接口：172.16.1.2，VLAN B”里面的IP是不是写错了，是“172.16.**2.1**”吧？

**Reply by isno** [reply]
谢谢，这里两处都写错了，我改改。

**Comment by dolphinlife** [comment]
VNI 字段（如图 3-25 所示），应该是如果3-18

**Reply by isno** [reply]
已修复，谢谢！

**Comment by lxchx** [comment]
> 只设计了 12 位 bit 存储 VLAN ID，导致一个 VLAN 子网内的设备数量局限在 4000 个左右

这里的说法是不是有误，vlan id是标识所属vlan的，所以应该是限制了二层网络的vlan只能有4000多个，而不是一个vlan内的设备只能有4000多个吧

**Reply by isno** [reply]
谢谢指正，这儿是我写错了。
